src/resources.py

```diff
@@ -43,15 +43,18 @@ def signal_handler(signum, frame):
 signal.signal(signal.SIGTERM, signal_handler)
 
 
-def _get_file_data_and_name(full_filename, content, enable_5xx, content_type=CONTENT_TYPE_TEXT):
+def _get_file_data_and_name(full_filename, content, enable_5xx, content_type=CONTENT_TYPE_TEXT, remove=False):
     if content_type == CONTENT_TYPE_BASE64_BINARY:
         file_data = base64.b64decode(content)
     else:
         file_data = content
 
-    if full_filename.endswith(".url"):
+    if full_filename.endswith(".url") and not remove:
```

Comment:
I think this change breaks removal as the returned filename would still contain the suffix `.url` which is not a part of the filename. To avoid running the request (and the script further down) on removals only setting `file_data` should be skipped, not `filename`.